### PR TITLE
bumped version to nNeo4j 4.2 and updated other dependencies

### PR DESCRIPTION
**Pull request by taseroth**
Updating 4.2 

* Neo4j dependencies updated to 4.2
* other available dependencies updates applied

### CODE REVIEW

File: src/test/java/example/TraverseDemoTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    private static final Config driverConfig = Config.builder().withoutEncryption().build();
+    private Neo4j embeddedDatabaseServer;
+
+    @BeforeAll
+    void initializeNeo4j() {
+        this.embeddedDatabaseServer = Neo4jBuilders.newInProcessBuilder()
+                .withProcedure(TraverseDemo.class)
+                .withFixture(new File(getClass().getResource("/movie.cypher").getPath()))
+                .build();
```
Comment (michael-simons) [inline src/test/java/example/TraverseDemoTest.java]:
```suggestion
        var sw = new StringWriter();
        try (var in = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream("/movie.cypher")))) {
            in.transferTo(sw);
            sw.flush();
        }

        this.embeddedDatabaseServer = Neo4jBuilders.newInProcessBuilder()
            .withProcedure(TraverseDemo.class)
            .withFixture(sw.toString())
            .build();
```

Don't assume a resource being a file. I should we should go with a good example here.

File: src/test/java/example/TraverseDemoTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Test
+    void findKeanuReevesCoActors() {
+
+        try(
+                Driver driver = GraphDatabase.driver(embeddedDatabaseServer.boltURI(), driverConfig);
+                Session session = driver.session()) {
+
+            List<String> names = session.run("match (keanu:Person {name:'Keanu Reeves'})-[*1..2]-(coactors:Person)\n" +
```
Comment (michael-simons) [inline src/test/java/example/TraverseDemoTest.java]: We can also use `var` types. 

```suggestion
            var names = session.run("match (keanu:Person {name:'Keanu Reeves'})-[*1..2]-(coactors:Person)\n" +
```

**Review by michael-simons — CHANGES_REQUESTED** [review]
I would change the test setup.

And with regards to PRs: Normally I would expect the dependency change and the new demo being 2 different things.

File: src/test/java/example/TraverseDemoTest.java (hunk 2 above)
Comment (taseroth) [reply src/test/java/example/TraverseDemoTest.java]: Thanks. Guess I'm still stuck in the 'old' world. Not doing enough development right now :-(

**Review by michael-simons — APPROVED** [review]

**Pull request merged**